src/nspawn/nspawn-bind-user.c

```diff
@@ -138,6 +146,7 @@ static int convert_user(
                                         SD_JSON_BUILD_PAIR_CONDITION(u->disposition >= 0, "disposition", SD_JSON_BUILD_STRING(user_disposition_to_string(u->disposition))),
                                         SD_JSON_BUILD_PAIR("homeDirectory", SD_JSON_BUILD_STRING(h)),
                                         SD_JSON_BUILD_PAIR("service", JSON_BUILD_CONST_STRING("io.systemd.NSpawn")),
+                                        SD_JSON_BUILD_PAIR_CONDITION(!!sh, "shell", SD_JSON_BUILD_STRING(sh)),
```

Comment:
`JSON_BUILD_PAIR_STRING_NON_EMPTY()` can be used.
At least, please drop `!!`.

Reply:
Handy, thanks!